src/vcpkg/base/system.cpp

```diff
@@ -325,7 +325,22 @@ namespace vcpkg
         return ProgramW6432;
     }
 
-    int get_num_logical_cores() { return std::thread::hardware_concurrency(); }
+    int get_concurrency()
+    {
+        static int concurrency = [] {
+            auto user_defined_concurrency = get_environment_variable("VCPKG_MAX_CONCURRENCY");
+            if (user_defined_concurrency)
+            {
+                return std::stoi(user_defined_concurrency.value_or_exit(VCPKG_LINE_INFO));
```

Comment:
what if `user_defined_concurrency` is not a positive integer?

Reply:
Then `get_concurrency()` will return a negative number; that's why it returns `int`. Agreed this is not a useful value, but this is not a code change from today -- the only use of `get_num_logical_cores()` was in `get_concurrency()` elsewhere which had this exact same semantic.